integration_tests/python_modules/dagster-k8s-test-infra/dagster_k8s_test_infra/cluster.py

```diff
@@ -282,7 +282,7 @@ def helm_postgres_url(helm_namespace):
 def dagster_instance(helm_postgres_url):  # pylint: disable=redefined-outer-name
 
     with tempfile.TemporaryDirectory() as tempdir:
-        with environ({"DAGSTER_HOME": tempdir}):
+        with environ({"DAGSTER_HOME": tempdir, "DAGSTER_DISABLE_TELEMETRY": True}):
```

Comment:
The necessity to do this in a bunch of places is clearly awful. I think that it would be way better if we could set this env var within either the call to `DagsterInstance.local_temp` or `DagsterInstance.temp_storage`, but those don't support context management, and we risk injecting the env var outside of scope. Not sure how changes to env vars are persisted after the end of a python program.

Reply:
maybe upload_logs should take in an instance and check settings on it, rather than checking an env var